controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -303,6 +309,34 @@ func (r *NodeNetworkConfigurationPolicyReconciler) deleteEnactmentForPolicy(poli
 	return nil
 }
 
+func (r *NodeNetworkConfigurationPolicyReconciler) syncPolicyMaxUnavailableCount(policy *nmstatev1beta1.NodeNetworkConfigurationPolicy) error {
+	policyKey := types.NamespacedName{Name: policy.GetName(), Namespace: policy.GetNamespace()}
+	instance := &nmstatev1beta1.NodeNetworkConfigurationPolicy{}
+	err := r.Client.Get(context.TODO(), policyKey, instance)
+	if err != nil {
+		return err
+	}
+	_, enactmentCountByCondition, err := enactment.CountByPolicy(r.Client, instance)
+	if err != nil {
+		return err
+	}
+	if instance.Status.UnavailableNodeCount != enactmentCountByCondition.Progressing() {
+		return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			err := r.Client.Get(context.TODO(), policyKey, instance)
+			if err != nil {
+				return err
+			}
+			_, enactmentCountByCondition, err = enactment.CountByPolicy(r.Client, instance)
+			if err != nil {
+				return err
+			}
+			instance.Status.UnavailableNodeCount = enactmentCountByCondition.Progressing()
+			return r.Client.Status().Update(context.TODO(), instance)
+		})
+	}
+	return nil
+}
+
```

Comment:
I think we can wrap all on RetryOnConflict

```suggestion
func (r *NodeNetworkConfigurationPolicyReconciler) syncPolicyMaxUnavailableCount(policy *nmstatev1beta1.NodeNetworkConfigurationPolicy, ) error {
       return retry.RetryOnConflict(retry.DefaultRetry, func() error {
	    policyKey := types.NamespacedName{Name: policy.GetName(), Namespace: policy.GetNamespace()}
	    instance := &nmstatev1beta1.NodeNetworkConfigurationPolicy{}
	    err := r.Client.Get(context.TODO(), policyKey, instance)
	    if err != nil {
		    return err
	    }
	    _, enactmentCountByCondition, err := enactment.CountByPolicy(r.Client, instance)
	    if err != nil {
                return err
	    }
	    if instance.Status.UnavailableNodeCount != enactmentCountByCondition.Progressing() {
	        instance.Status.UnavailableNodeCount = enactmentCountByCondition.Progressing()
		return r.Client.Status().Update(context.TODO(), instance)
	    })
	}
	return nil
}

```

Reply:
This looks better indeed. Updated